google/resource_compute_attached_disk.go

```diff
@@ -0,0 +1,282 @@
+
+	return nil
+}
+
+func resourceAttachedDiskDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("ZOMG destroying")
+	config := meta.(*Config)
+
+	project, err := getProjectForAttachedDisk(d, config)
+	if err != nil {
+		return err
+	}
+	zone, err := getZoneForAttachedDisk(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := GetResourceNameFromSelfLink(d.Get("instance").(string))
+	diskName := GetResourceNameFromSelfLink(d.Get("disk").(string))
+
+	instance, err := config.clientCompute.Instances.Get(project, zone, instanceName).Do()
+	if err != nil {
+		return err
+	}
+
+	// Confirm the disk is still attached before making the call to detach it. If the disk isn't listed as an attached
+	// disk on the compute instance then return as though the delete call succeed since this is the desired state.
+	ad := findDiskByName(instance.Disks, diskName)
+	if ad == nil {
+		return nil
+	}
+
+	log.Printf("ZOMG calling detach")
+	op, err := config.clientCompute.Instances.DetachDisk(project, zone, instanceName, ad.DeviceName).Do()
+	if err != nil {
+		return err
+	}
+
+	waitErr := computeSharedOperationWaitTime(config.clientCompute, op, project,
+		int(d.Timeout(schema.TimeoutDelete).Minutes()), fmt.Sprintf("Detaching disk from %s", instanceName))
+	if waitErr != nil {
+		return waitErr
+	}
+
+	return nil
+}
+
+func resourceAttachedDiskImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
+
+	err := parseImportId(
+		[]string{"projects/(?P<project>[^/]+)/zones/(?P<zone>[^/]+)/instances/[^/]+",
+			"(?P<project>[^/]+)/(?P<zone>[^/]+)/[^/]+"}, d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	// In all acceptable id formats the actual id will be the last in the path
+	id := GetResourceNameFromSelfLink(d.Id())
+	d.SetId(id)
+
+	IDParts := strings.Split(d.Id(), ":")
+	if len(IDParts) != 2 {
+		return nil, fmt.Errorf("unable to determine attached disk id - id should be 'google_compute_instance.name:google_compute_disk.name'")
+	}
+	d.Set("instance", IDParts[0])
+	d.Set("disk", IDParts[1])
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func findDiskByName(disks []*compute.AttachedDisk, id string) *compute.AttachedDisk {
+	for _, disk := range disks {
+		if compareSelfLinkOrResourceName("", disk.Source, id, nil) {
+			return disk
+		}
+	}
+
+	return nil
+}
+
+// getZoneForAttachedDisk prioritizes the zone defined by the compute instance self link before standard logic
+func getZoneForAttachedDisk(d *schema.ResourceData, c *Config) (string, error) {
+	attachedInstance := d.Get("instance").(string)
+
+	zone, err := GetPathVariableFromSelfLink(attachedInstance, "zone")
+	if err == nil {
+		return zone, nil
+	}
+
+	// If zone can't be inferred from the compute instance self link, fall back to project
+	zone, err = getZone(d, c)
+	if err != nil {
+		return "", fmt.Errorf("%s to inherit from the attached instance compute use `self_link` instead of `name`", err)
```

Comment:
It's layering on top of the [error thrown by the getZone method](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/regional_utils.go#L34). So it would read: "Cannot determine zone: set in this resource, or set provider-level zone. to inherit from the attached instance compute use `self_link` instead of `name`"

I dropped the punctuation as suggested in https://github.com/golang/go/wiki/CodeReviewComments#error-strings but a comma could probably go a long way. Also since we already use full sentences in our error strings I could just change it into a separate sentence. Thoughts?